### ISSUE TITLE
Incorrectly set number of posts in loop

### ISSUE THREAD

**Issue by danieltj27** [danieltj27/Custom-Archives#9]
In issue #6, it addressed the fact that if you had no posts for a particular post type, if you had a post loop on the page template for the custom archive, it'd say no posts found rather than showing the proper page content (as the number of posts was set to 0).

In commit 65e9987, this was attempted to be fixed but it's set to aggressively set the number of posts to 1 every time. This needs fixing so it counts the total posts first, if 0 then reset to 1, so it still works.

So roughly on line 742, change that line to:

```
    $wp_query->post_count = count( $wp_query->query_vars['archive_posts'] );
```

This is untested though, it may not work so needs thorough testing here.

**Comment by danieltj27** [comment]
This should be an easy fix, but a temporary fix for now is to turn the post loop setting off. Might not be ideal in situations where it's needed but will get a fix done soon.

**Comment by danieltj27** [comment]
Both `$wp_query->post_count` and `$wp_query->posts_per_page` need setting to 1, this is the only fix I think. Considering the page should be treated as a normal page anyway, there's no (or at least shouldn't be) any instances where you'd want the normal post loop for the current page content.

The solution here might be to make the `$wp_query` variable filterable and sent these values to 1 so it's treated as if it was it's own normal page and remove this setting.

**Comment by danieltj27** [comment]
Finally fixed in commit 91c5dde.
